include/extractor/guidance/statistics_handler.hpp

```diff
@@ -0,0 +1,94 @@
+#ifndef OSRM_EXTRACTOR_GUIDANCE_STATISTICS_HANDLER_HPP_
+#define OSRM_EXTRACTOR_GUIDANCE_STATISTICS_HANDLER_HPP_
+
+#include "extractor/guidance/intersection.hpp"
+#include "extractor/guidance/intersection_handler.hpp"
+#include "extractor/guidance/turn_instruction.hpp"
+
+#include "util/log.hpp"
+
+#include <mutex>
+#include <unordered_map>
+
+#include <cstdint>
+
+namespace osrm
+{
+namespace extractor
+{
+namespace guidance
+{
+
+// Unconditionally runs over all intersections and gathers statistics for
+// instruction turn types and direction modifiers (see turn_instruction.hpp).
+class StatisticsHandler final : public IntersectionHandler
+{
+  public:
+    StatisticsHandler(const IntersectionGenerator &intersection_generator,
+                      const util::NodeBasedDynamicGraph &node_based_graph,
+                      const EdgeBasedNodeDataContainer &node_data_container,
+                      const std::vector<util::Coordinate> &coordinates,
+                      const util::NameTable &name_table,
+                      const SuffixTable &street_name_suffix_table)
+        : IntersectionHandler(node_based_graph,
+                              node_data_container,
+                              coordinates,
+                              name_table,
+                              street_name_suffix_table,
+                              intersection_generator)
+    {
+    }
+
+    ~StatisticsHandler() override final
+    {
+        // Todo: type and modifier to string
+
+        util::Log() << "Assigned turn instruction types";
+
+        for (const auto &kv : type_hist)
+            util::Log() << internalInstructionTypeToString(kv.first) << ": " << kv.second;
```

Comment:
Two things here:

1. It would be cool of if we only print instructions that have a count bigger then zero.
2. A relative number `left turn: 1337 (42%)` would help quantify.